src/vm/hints/hint_utils.rs

```diff
@@ -47,6 +46,42 @@ fn get_address_from_reference(
     None
 }
 
+fn compute_addr_from_one_offset(
+    relocatable: Relocatable,
+    hint_reference: &HintReference,
+) -> Option<MaybeRelocatable> {
+    if hint_reference.offset1.is_negative()
+        && relocatable.offset < hint_reference.offset1.abs() as usize
+    {
+        return None;
+    }
+    Some(MaybeRelocatable::from((
+        relocatable.segment_index,
+        (relocatable.offset as i32 + hint_reference.offset1) as usize,
+    )))
+}
+
+fn compute_addr_from_two_offsets(
+    relocatable: Relocatable,
+    hint_reference: &HintReference,
+) -> Option<MaybeRelocatable> {
+    let addr = if hint_reference.offset1.is_negative()
+        && relocatable.offset < hint_reference.offset1.abs() as usize
+    {
+        return None;
```

Comment:
I'm not quite sure why we do this, could you explain this to me?

Reply:
This is just for checking that when we subtract the hint reference offset to the offset of the memory cell, we don't go outside the memory segment. I discussed this with @fmoletta and she told me this should be the expected behaviour

Reply:
Does `relocatable.offset` indicate the max offset for the segment?